src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1475,12 +1475,61 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                         _ => None,
                     } {
                         debug!("id ={:?} span = {:?} name = {:?}", node_id, span, name);
-                        self.tcx.struct_span_lint_node(
+                        let mut err = self.tcx.struct_span_lint_node(
                             lint::builtin::UNUSED_LIFETIMES,
                             id,
                             span,
                             &format!("lifetime parameter `{}` never used", name)
-                        ).emit();
+                        );
+                        if let Some(parent_def_id) = self.tcx.parent(def_id) {
```

Comment:
There’s a lot going on here. Consider putting some or all of this code in a helper function.

Reply:
Yeah... @zackmdavis said as much in the PR intro of course :) it would be nice to see if we could extract a helper or something at some point though.

Reply:
Right.